src/main/java/app/model/Map.java

```diff
@@ -257,4 +257,27 @@ private boolean clearSpot(Vector v)
         }
         return true;
     }
+
+
+    public void capture(Agent currentAgent)
+    {
+        if(currentAgent.getTeam() == Team.GUARD)
+            for(Agent otherAgent: agents){
+                if (otherAgent.getTeam() !=currentAgent.getTeam()){
+                    double dis = currentAgent.getPosition().dist(otherAgent.getPosition());
+                    if(dis <= currentAgent.getRadius() + otherAgent.getRadius()){
```

Comment:
This checks if they have exactly touched, but we discussed having a buffer (like an extra 5 or so) that gives the agent a "Capture Radius" that works like an arms length (or taser length) for them to capture in ...